File: Array /125. Valid Palindrome/125. Valid Palindrome.cpp
bool isPalindrome(char* s) {
    int len=strlen(s);
    int end=len;int start=0;
    
    for(int i=0;i<len;i++){
        if(s[i]>='A'&&s[i]<='Z'){
            s[i]=s[i]+32;
        }
    }

    while(start<end){
        while( start<end && (s[start]<'a' || s[start]>'z') && (s[start]<'0' || s[start]>'9' )){
            start++;
        }
         while( start<end && (s[end]<'a' || s[end]>'z') && (s[end]<'0' || s[end]>'9' )){
            end--;
        }

        if(s[start]==s[end]){
            start++;
            end--;
        }else{
            return false;
        }
        
    }
    return true;
}

File: Array /125. Valid Palindrome/125ValidPalindrome.py
class Solution {
public:
    bool isPalindrome(string s) {
        int front = 0;
        int back = s.length()-1;
        
        while (front < back) {
            while (front < s.length()-1 && !isalnum(s[front])) front++;
            while (back > 0 && !isalnum(s[back])) back--;
            
            if (tolower(s[front]) != tolower(s[back]) 
                && isalnum(s[front]) && isalnum(s[back])){
                return false;
            }
            front++;
            back--;
        }
        
        return true;
    }
};
